crates/uv/src/commands/project/init.rs

```diff
@@ -41,79 +43,185 @@ pub(crate) async fn init(
+                    path.display().cyan()
+                );
+            }
+
+            // Default to the directory name if a name was not provided.
+            let name = match name {
+                Some(name) => name,
+                None => {
+                    let name = path
+                        .file_name()
+                        .and_then(|path| path.to_str())
+                        .context("Missing directory name")?;
+
+                    PackageName::new(name.to_string())?
+                }
+            };
+
+            init_project(
+                &path,
+                &name,
+                package,
+                project_kind,
+                no_readme,
+                no_pin_python,
+                python,
+                no_workspace,
+                python_preference,
+                python_downloads,
+                connectivity,
+                native_tls,
+                cache,
+                printer,
+            )
+            .await?;
 
-    // Make sure a project does not already exist in the given directory.
-    if path.join("pyproject.toml").exists() {
-        let path = std::path::absolute(&path).unwrap_or_else(|_| path.simplified().to_path_buf());
-        anyhow::bail!(
-            "Project is already initialized in `{}` (`pyproject.toml` file exists)",
-            path.display().cyan()
-        );
+            // Create the `README.md` if it does not already exist.
+            if !no_readme {
+                let readme = path.join("README.md");
+                if !readme.exists() {
+                    fs_err::write(readme, String::new())?;
+                }
+            }
+
+            match explicit_path {
+                // Initialized a project in the current directory.
+                None => {
+                    writeln!(printer.stderr(), "Initialized project `{}`", name.cyan())?;
+                }
+                // Initialized a project in the given directory.
+                Some(path) => {
+                    let path = std::path::absolute(&path)
+                        .unwrap_or_else(|_| path.simplified().to_path_buf());
+                    writeln!(
+                        printer.stderr(),
+                        "Initialized project `{}` at `{}`",
+                        name.cyan(),
+                        path.display().cyan()
+                    )?;
+                }
+            }
+        }
     }
 
-    // Default to the directory name if a name was not provided.
-    let name = match name {
-        Some(name) => name,
-        None => {
-            let name = path
-                .file_name()
-                .and_then(|path| path.to_str())
-                .context("Missing directory name")?;
+    Ok(ExitStatus::Success)
+}
 
-            PackageName::new(name.to_string())?
+#[allow(clippy::fn_params_excessive_bools)]
+async fn init_script(
+    script_path: &Path,
```

Comment:
You generally want to avoid `&PathBuf`, similar to how you would avoid `&String`. The reasoning is similar to the previous comments around `&Option<T>` vs. `Option<&T>`. If you have a `PathBuf`, you can always get an `&Path`; but if you have a `Path`, you can't convert to `&PathBuf` without allocating.

Reply:
I see, ```Path``` is to ```PathBuf``` what ```&str``` is to ```String``` clarifies this well, thanks!